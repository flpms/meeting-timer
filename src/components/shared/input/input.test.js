import React from 'react';
import { render, unmountComponentAtNode} from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import Input from './index.js';

describe('Input', () => {
  it('renders input without crashing', () => {
    const div = document.createElement('div');
    render(<Input />, div);
    unmountComponentAtNode(div);
  });

  it('expect test something', () => {
    mount(<Input label="Theme"
      keyId="theme"
      onChange={(e) => e} />)
  });

  it('[snapshot] input when hovered', () => {
    const component = renderer.create(<Input label="Theme"
        keyId="theme"
        onChange={ (e) => e } />,
    );
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
    tree.children[1].props.onMouseEnter();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
    tree.children[1].props.onMouseLeave();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
