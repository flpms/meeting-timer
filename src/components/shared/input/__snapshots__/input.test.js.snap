// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input [snapshot] input when hovered 1`] = `
<div
  className="input"
>
  <label
    htmlFor="theme"
  >
    Theme
  </label>
  <input
    className="text"
    id="theme"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`Input [snapshot] input when hovered 2`] = `
<div
  className="input add-label-effect"
>
  <label
    htmlFor="theme"
  >
    Theme
  </label>
  <input
    className="text"
    id="theme"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`Input [snapshot] input when hovered 3`] = `
<div
  className="input"
>
  <label
    htmlFor="theme"
  >
    Theme
  </label>
  <input
    className="text"
    id="theme"
    onChange={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="text"
    value=""
  />
</div>
`;
